bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java

```diff
@@ -491,7 +491,8 @@ private OutputStream addPreSteps(ProcessingStepHandler handler, IArtifactDescrip
 		addChecksumVerifiers(descriptor, downloadChecksumSteps, skipChecksums, IArtifactDescriptor.DOWNLOAD_CHECKSUM);
 		if (downloadChecksumSteps.isEmpty() && !isLocal()) {
 			LogHelper.log(new Status(IStatus.WARNING, Activator.ID,
-					NLS.bind(Messages.noDigestAlgorithmToVerifyDownload, descriptor.getArtifactKey())));
```

Comment:
Notice that this call is guarded by `!isLocal()" because of course if the repository is local, this is not a download.